ddtrace/contrib/httplib/__init__.py

```diff
@@ -9,21 +9,15 @@
     patch(httplib=True)
 
     # Python 2
-    from ddtrace import Pin
     import httplib
     import urllib
 
-    # Use a Pin to specify metadata for all http requests
-    Pin.override(httplib, service='httplib')
     resp = urllib.urlopen('http://www.datadog.com/')
 
     # Python 3
-    from ddtrace import Pin
     import http.client
     import urllib.request
 
-    # Use a Pin to specify metadata for all http requests
-    Pin.override(http.client, service='httplib')
     resp = urllib.request.urlopen('http://www.datadog.com/')
```

Comment:
I think the problem here is that because no service name will be set, we'll default to `unnamed-python-service` in new agents or get rejected in older agents.

We might instead need to do:

```python
with tracer.trace('main', service='my-operation'):
    resp = urllib.request.urlopen(...)
```

Along with a description saying that `httplib` must be the child of a span with a service name, either manually or via an integration.

As well, I do believe from that other issue that `Pin` works if you pin an instance of a client, but not if you pin the library.

Reply:
Makes sense to add the instructions on starting a trace with a service name before spans from the `httplib` integration are handled.

As for the `Pin`, yes it is pulled from the instance in the integration:

https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/contrib/httplib/patch.py#L26

https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/contrib/httplib/patch.py#L52

But it's also the case we only use the `pin.tracer` and not any metadata (e.g., `service_name`). 

https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/contrib/httplib/patch.py#L58

This documentation change was to reflect the current implementation as not respecting any meaningful overrides a developer might make on the instance.

Reply:
👍